world/eqemu_api_world_data_service.cpp

```diff
@@ -270,8 +272,64 @@ void EQEmuApiWorldDataService::get(Json::Value &r, const std::vector<std::string
 	}
 	if (m == "get_reload_types") {
 		getReloadTypes(r);
-	}
+	}				
 	if (m == "reload") {
 		reload(r, args);
 	}
+	if (m == "get_guild_details") {
+		callGetGuildDetails(r, args);
+	}
+}
+
+void EQEmuApiWorldDataService::callGetGuildDetails(Json::Value& response, const std::vector<std::string>& args)
+{
+
+	std::string command = !args[1].empty() ? args[1] : "";
+	if (command.empty()) {
+		return;
+	}
+	Json::Value row;
+
+	auto guild_id = Strings::ToUnsignedInt(command);
+	if (!guild_id) {
+		row = "useage is: api get_guild_details ### where ### is a valid guild id.";
+		return;
+	}
+	auto guild = guild_mgr.GetGuildByGuildID(guild_id);
+	if (!guild) {
+		row = fmt::format("Could not find guild id {}", guild_id);
+		return;
+	}
+
+	row["guild_id"]    = command;
+	row["guild_name"]  = guild->name;
+	row["leader_id"]   = guild->leader;
+	row["min_status"]  = guild->minstatus;
+	row["motd"]        = guild->motd;
+	row["motd_setter"] = guild->motd_setter;
+	row["url"]         = guild->url;
+	row["channel"]     = guild->channel;
+
+	for (int i = 1; i <= 8; i++) {
+		row["Ranks"][i] = guild->rank_names[i].c_str();
+	}
+
+	for (int i = 1; i <= 30; i++) {
+		row["Functions"][i]["db_id"]		= guild->functions[i].id;
+		row["Functions"][i]["perm_id"]		= guild->functions[i].perm_id;
+		row["Functions"][i]["guild_id"]		= guild->functions[i].guild_id;
+		row["Functions"][i]["perm_value"]	= guild->functions[i].perm_value;
+	}
+
+	row["Tribute"]["Favor"]			 = guild->tribute.favor;
```

Comment:
These should all be lowercase and `snake_case`

Reply:
```
row["tribute"]["favor"]
```

Reply:
updated